docs/en/pact-functions.rst

```diff
@@ -734,6 +745,24 @@ chain using automated SPV endorsement-based dispatch.
    (yield { "amount": 100.0 })
    (yield { "amount": 100.0 } "some-chain-id")
 
+zip
+~~~
+
+*f* ``x:<a> y:<b> -> <c>`` *list1* ``[<a>]`` *list2* ``[<b>]``
+*→* ``[<c>]``
+
+Combine two lists with some function f, into a new list, the length of
+which is the length of the shortest list.
+
+.. code:: lisp
+
+   pact> (zip (+) [1 2 3 4] [4 5 6 7])
+   [5 7 9 11]
+   pact> (zip (-) [1 2 3 4] [4 5 6])
+   [-3 -3 -3]
+   pact> (zip (+) [1 2 3] [4 5 6 7])
+   [5 7 9]
+
```

Comment:
a classic `zip` example pairing values in a product (object) would be good here. There are redundant examples.

Reply:
I do not believe zip supports this.

Reply:
> I do not believe zip supports this
```
  (zip (lambda (x y) { 'x: x, 'y: y }) [...] [...]
```
?

Reply:
Oh woops. Sure I could add an example.